Trivy >=v0.23.0 panics on interface conversion when scanning some images

## Description

`trivy image` is failing for a small subset of images on both v0.23.0 and v0.24.0 with `panic: interface conversion: interface {} is convert.jsonObj, not []interface {}`

## Output of run with `-debug`:

<details>
<summary>Debug output</summary>

```
2022-02-25T15:59:57.652-0500	DEBUG	Severities: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
2022-02-25T15:59:57.696-0500	DEBUG	cache dir:  <removed for privacy>
2022-02-25T15:59:57.697-0500	DEBUG	DB update was skipped because the local DB is the latest
2022-02-25T15:59:57.701-0500	DEBUG	DB Schema: 2, UpdatedAt: 2022-02-25 18:07:08.075847137 +0000 UTC, NextUpdate: 2022-02-26 00:07:08.075846937 +0000 UTC, DownloadedAt: 2022-02-25 20:13:14.824499 +0000 UTC
2022-02-25T15:59:57.701-0500	DEBUG	Vulnerability type:  [os library]
2022-02-25T15:59:58.744-0500	DEBUG	Image ID: sha256:4c5afb543ccad90e0d4151c66d30b148bb2fcbe9deb9bb9a8a2f2250aa40e13f
2022-02-25T15:59:58.744-0500	DEBUG	Diff IDs: [sha256:80580270666742c625aecc56607a806ba343a66a8f5a7fd708e6c4e4c07a3e9b sha256:3fd9df55318470e88a15f423a7d2b532856eb2b481236504bf08669013875de1 sha256:7a694df0ad6cc5789a937ccd727ac1cda528a1993387bf7cd4f3c375994c54b6 sha256:52691343cabbb2d2bd7025bb52443a4496f49354213c4692f5f206a8b85b43b0 sha256:302986ba9bc153287f5d745c78860c5e22a444201be81f632360ff30ff24214f sha256:2fbaf896f6b7d4eba4f318c0b68ca6afeeec3bd894bee4577b6cdd625bd7e5f7 sha256:fdea6f88dd37ce9009aeb728d53ed65b7c9d671a5496a981ab43d00059618241 sha256:23550b588629b464cc331c9101dc0a54957d26faaca91db245f90c3b2660f82a sha256:262e7ba7af064c97b26d1eae6139e214759cd272d3017d3e94a69dc21068404c sha256:8f50cfc516c2b95c2c50d7ec3f6b4275db32b750b710ea666559c9864aa725e9 sha256:4e845e4555da87965f1d239d348c1a342d894cedef247345b9a4497f4d91e6fc sha256:7e0a3ca07b02781678cf5c7bb6490548cd410700cfe58f82d236586ca43000c8 sha256:8b296afbe96ff9c879472db328e6fc236eaca12a9a335f47fbe2997b96880488 sha256:4c8bf176efdbe957a554af1c1e6cdeba878aecf553f630eefe0c330d65a7ae13 sha256:0f2fbfe8737c98d47e23925a0afed7e39a9c6fcc7bd7834dfcff663f89f4dd27 sha256:6b2ee8d67ee408e8cefdf429f05eb68e2210434ce01cf7406c75a067030ce205 sha256:0d1583daa934496a5652e18ce6a8374ac5e1827e1db502c2e165e1bbfa29399d sha256:be5f62ac2a2f45dc74b556eef3ff5131ec5f1f615e976ad4758ac2e46db5677c sha256:a936d718a00a56de36976eb499298cfb9d956dbf69256778cdd4a5dcaae20f54 sha256:fc5a5161b215af7de983dbb267837edf09c8bcd2f513a69f016195e9e9279a4e sha256:35d0626ed3d11825ac3828f91ba5c7724177409bc1c1e2093c3226d8459e30ab sha256:e503d57a6f0d8ef50c164a6281787478002274f49b65b70de9a96e7c0930321f sha256:8a7b6b8c68fec568ab8d6694af7bc5aaa50d4ecf71176b911579ce03bf58de94 sha256:4377e10c6630913fc5d6a4b8ec231edb1649118129615bf9fbc7c573863aebb8 sha256:0f1fed2736565782a96ff9e20f678da539f7dfd887891d4c8e7701ffeee254ae sha256:604f2ee40a2983a5002411282fb26b1008467573ac2ef1567abaf93ca1452c6d sha256:acd05a25f339e69539f5a3059b5251ccc367669c9acb3221d7e9698215d62ad4 sha256:ac822818f8fc1131cba547c64592b91d4fc12903f38dbebcead98741759cac58 sha256:ada9d3a17e68fa163cbf47bebb7b352b3f3e42084b7cfa4175a3387339c3e052 sha256:08027d7a011a14cba952939f8b75018b5bb01eb9838a3d3860f4e011ac0cb27c sha256:5eefa08f5332a7d01b8085c340d88cf2bf3a091863c36455ef504629c76f29ef sha256:00ac2713794fe0b89dcbfe256fef95c9a1b893cdc529828883f94f2042c82fff sha256:4656487917d5e6a9bb55fe5aa6c9be2baa1243a5568032723ddc2e3338c012b9 sha256:575bf3a77878d2fe93fee566fd5b8573f0e03043f7644113d6fc7af409be075c sha256:50d04bea4034342fed9a1ad861ce84921906b4bb2d202e82da930a40621cef06 sha256:98b63cc4b3ed5081f815e8faef44420f7ce2ef2a5ce2a856f722dcaf15eb5b2b sha256:1e35017d4b9902bed59880bf331ccaa0ee6256230a8771131b26d31485e5b666 sha256:bc4c30fa3e5e9150cac77f80adaf4d205c583eaf56a589cc0b505a368e664392 sha256:c28f20415d3d631b119a7d69c46c4fb23cf0d647eb31610b4268efbbba266df0]
2022-02-25T15:59:58.752-0500	DEBUG	Missing image ID: sha256:4c5afb543ccad90e0d4151c66d30b148bb2fcbe9deb9bb9a8a2f2250aa40e13f
2022-02-25T15:59:58.753-0500	DEBUG	Missing diff ID: sha256:08027d7a011a14cba952939f8b75018b5bb01eb9838a3d3860f4e011ac0cb27c
2022-02-25T15:59:58.753-0500	DEBUG	Missing diff ID: sha256:e503d57a6f0d8ef50c164a6281787478002274f49b65b70de9a96e7c0930321f
2022-02-25T15:59:58.753-0500	DEBUG	Missing diff ID: sha256:0f1fed2736565782a96ff9e20f678da539f7dfd887891d4c8e7701ffeee254ae
2022-02-25T15:59:58.753-0500	DEBUG	Missing diff ID: sha256:fc5a5161b215af7de983dbb267837edf09c8bcd2f513a69f016195e9e9279a4e
2022-02-25T15:59:58.753-0500	DEBUG	Missing diff ID: sha256:35d0626ed3d11825ac3828f91ba5c7724177409bc1c1e2093c3226d8459e30ab
2022-02-25T15:59:58.754-0500	DEBUG	Missing diff ID: sha256:acd05a25f339e69539f5a3059b5251ccc367669c9acb3221d7e9698215d62ad4
2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/array_comment_2.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/array_comment_2.hcl): convert hcl2 to bytes: parse config: [:4,5-6: Missing item separator; Expected a comma to mark the beginning of the next item.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/array_comment_2.hcl): At 4:5: error parsing list, expected comma or list end, got: STRING


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/key_without_value.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/key_without_value.hcl): convert hcl2 to bytes: parse config: [:1,1-4: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/key_without_value.hcl): At 2:1: key 'foo' expected start of object ('{') or assignment ('=')


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/git_crypt.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/git_crypt.hcl): convert hcl2 to bytes: parse config: [:1,1-2: Invalid character; This character is not used within the language. :1,1-2: Argument or block definition required; An argument or block definition is required here.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/git_crypt.hcl): At 1:1: unexpected null character (0x00)


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/missing_braces.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/missing_braces.hcl): convert hcl2 to bytes: parse config: [:3,22-23: Invalid character; This character is not used within the language. :3,22-23: Missing newline after argument; An argument definition must end with a newline.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/missing_braces.hcl): At 3:22: illegal char


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value.hcl): convert hcl2 to bytes: parse config: [:2,8-3,1: Invalid expression; Expected the start of an expression, but found an invalid expression token.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value.hcl): At 4:1: object expected closing RBRACE got: EOF


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value2.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value2.hcl): convert hcl2 to bytes: parse config: [:3,8-4,1: Invalid expression; Expected the start of an expression, but found an invalid expression token.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value2.hcl): At 5:1: object expected closing RBRACE got: EOF


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_without_value.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_without_value.hcl): convert hcl2 to bytes: parse config: [:2,3-6: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_without_value.hcl): At 3:1: key 'bar' expected start of object ('{') or assignment ('=')


2022-02-25T15:59:59.335-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value3.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value3.hcl): convert hcl2 to bytes: parse config: [:2,8-3,1: Invalid expression; Expected the start of an expression, but found an invalid expression token.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/object_key_assign_without_value3.hcl): At 3:3: Unknown token: 3:3 IDENT baz


2022-02-25T15:59:59.336-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/types.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/types.hcl): convert hcl2 to bytes: parse config: [:4,1-4: Attribute redefined; The argument "foo" was already set at :1,1-4. Each argument may be set only once. :5,1-4: Attribute redefined; The argument "bar" was already set at :2,1-4. Each argument may be set only once. :6,1-4: Attribute redefined; The argument "foo" was already set at :1,1-4. Each argument may be set only once. :7,1-4: Attribute redefined; The argument "bar" was already set at :2,1-4. Each argument may be set only once.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/types.hcl): At 5:7: root.bar: unknown type *ast.LiteralType


2022-02-25T15:59:59.336-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object.hcl): convert hcl2 to bytes: parse config: [:1,11-12: Unclosed configuration block; There is no closing brace for this block before the end of the file. This may be caused by incorrect brace nesting elsewhere in this file.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object.hcl): At 3:2: object expected closing RBRACE got: EOF


2022-02-25T15:59:59.336-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object_2.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object_2.hcl): convert hcl2 to bytes: parse config: [:1,29-32: Argument definition required; A single-line block definition can contain only a single argument. If you meant to define argument "a", use an equals sign to assign it a value. To define a nested block, place it on a line of its own within its parent block.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/parser/test-fixtures/unterminated_object_2.hcl): At 7:2: object expected closing RBRACE got: EOF


2022-02-25T15:59:59.339-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/hcl/test-fixtures/types.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/hcl/test-fixtures/types.hcl): convert hcl2 to bytes: parse config: [:4,1-4: Attribute redefined; The argument "foo" was already set at :1,1-4. Each argument may be set only once. :5,1-4: Attribute redefined; The argument "bar" was already set at :2,1-4. Each argument may be set only once. :6,1-4: Attribute redefined; The argument "foo" was already set at :1,1-4. Each argument may be set only once. :7,1-4: Attribute redefined; The argument "bar" was already set at :2,1-4. Each argument may be set only once.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/hcl/test-fixtures/types.hcl): At 5:7: root.bar: unknown type *ast.LiteralType


2022-02-25T15:59:59.339-0500	DEBUG	Analysis error: unable to decode JSON (go/src/github.com/hashicorp/hcl/json/parser/test-fixtures/bad_input_128.json): invalid character ':' looking for beginning of object key string
2022-02-25T15:59:59.339-0500	DEBUG	Analysis error: unable to decode JSON (go/src/github.com/hashicorp/hcl/json/parser/test-fixtures/bad_input_tf_8110.json): invalid character 'e' after object key:value pair
2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/block_assign.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/block_assign.hcl): convert hcl2 to bytes: parse config: [:1,21-22: Missing newline after argument; An argument definition must end with a newline.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/block_assign.hcl): At 1:21: expected: IDENT | STRING got: LBRACE


2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/git_crypt.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/git_crypt.hcl): convert hcl2 to bytes: parse config: [:1,1-2: Invalid character; This character is not used within the language. :1,1-2: Argument or block definition required; An argument or block definition is required here.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/git_crypt.hcl): At 1:1: unexpected null character (0x00)


2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/multiline_bad.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_bad.hcl): convert hcl2 to bytes: parse config: [:1,7-8: Invalid expression; Expected the start of an expression, but found an invalid expression token. :2,1-4: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value. :3,1-4: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value. :4,1-4: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_bad.hcl): At 1:8: heredoc expected second '<', didn't see it


2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/multiline_literal.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_literal.hcl): convert hcl2 to bytes: parse config: [:1,27-2,1: Invalid multi-line string; Quoted strings may not be split over multiple lines. To produce a multi-line string, either use the \n escape to represent a newline character or use the "heredoc" multi-line template syntax. :1,27-2,1: Unterminated template string; No closing marker was found for the string.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_literal.hcl): At 2:9: literal not terminated


2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/multiline_no_marker.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_no_marker.hcl): convert hcl2 to bytes: parse config: [:1,7-8: Invalid expression; Expected the start of an expression, but found an invalid expression token.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/multiline_no_marker.hcl): At 1:9: zero-length heredoc anchor


2022-02-25T15:59:59.340-0500	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/github.com/hashicorp/hcl/test-fixtures/nested_provider_bad.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/github.com/hashicorp/hcl/test-fixtures/nested_provider_bad.hcl): convert hcl2 to bytes: parse config: [:2,20-21: Missing newline after argument; An argument definition must end with a newline.]
	* unable to parse HCL1 (go/src/github.com/hashicorp/hcl/test-fixtures/nested_provider_bad.hcl): At 2:20: expected: IDENT | STRING got: LBRACE


panic: interface conversion: interface {} is convert.jsonObj, not []interface {}

goroutine 278 [running]:
github.com/tmccombs/hcl2json/convert.(*converter).convertBlock(0x0, 0xc0011f4120, 0x1)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:148 +0x2bd
github.com/tmccombs/hcl2json/convert.(*converter).convertBody(0x0, 0xc000f72e70)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:80 +0x9b
github.com/tmccombs/hcl2json/convert.ConvertFile(0x0, {0x40})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:68 +0x6f
github.com/tmccombs/hcl2json/convert.File(0x0, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:37 +0x1d
github.com/tmccombs/hcl2json/convert.Bytes({0xc0011e8600, 0x0, 0x2d26fe0}, {0x0, 0x200}, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:27 +0x78
github.com/aquasecurity/fanal/config/parser/hcl2.Unmarshal({0xc0011e8600, 0xc0010c9af0, 0xc0010c9af0}, {0x25ba820, 0xc0010c9b10})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220221141823-58a5d668e45f/config/parser/hcl2/hcl2.go:11 +0x33
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.analyze({0x0}, {{0x0, 0x0}, {0xc000945ac0, 0x3c}, {0x2d69448, 0xc0011d22a0}, {0x5e06580, 0xc0010c9af0}, {0x0}})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220221141823-58a5d668e45f/analyzer/config/hcl/hcl.go:61 +0xdf
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.Analyze({0x3df9de8}, {0x2d5a098, 0xc00059d0e0}, {{0x0, 0x0}, {0xc000945ac0, 0x3c}, {0x2d69448, 0xc0011d22a0}, {0x5e06580, ...}, ...})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220221141823-58a5d668e45f/analyzer/config/hcl/hcl.go:36 +0x6b
github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile.func1({0x2d5bf00, 0x0}, {0x2d5c018, 0xc0010c9af0})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220221141823-58a5d668e45f/analyzer/analyzer.go:253 +0x275
created by github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220221141823-58a5d668e45f/analyzer/analyzer.go:248 +0x3d2

```
</details>

## Output of `trivy -v`:

```
Version: 0.24.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-02-25 18:07:08.075847137 +0000 UTC
  NextUpdate: 2022-02-26 00:07:08.075846937 +0000 UTC
  DownloadedAt: 2022-02-25 20:13:14.824499 +0000 UTC
```

## Additional details (base image name, container registry info...):

The image I'm currently testing with is an amd64 container built with docker 18.03.1-ce, that's mostly meant to serve as a toolbox, so it's just a bunch of binaries. There are a few others that fail, generally similar images (similar as in mostly the same set of software, just different versions of the same file).
I can scan the base image with no issues.

--------

Originally posted as https://github.com/aquasecurity/trivy/discussions/1675
<details>
  <summary>Discussion post content</summary>
  
  

<div type='discussions-op-text'>

<sup>Originally posted by **mxmeinhold** February  4, 2022</sup>
I'm running into `trivy image` failures for a few images (only a very small proportion).  The images that are failing are `application/vnd.docker.distribution.manifest.v2+json`. They're well formed and pass `crane validate`, but I'm not familiar with the docker config object so maybe there's something unusual there? I don't have an image that I can put on github at the moment, but I can make an example over the weekend or early next week if the error alone doesn't point to an obvious cause.

This worked this morning on v0.22.0, and started failing after upgrading to v0.23.0

Output:
```
panic: interface conversion: interface {} is convert.jsonObj, not []interface {}

goroutine 730 [running]:
github.com/tmccombs/hcl2json/convert.(*converter).convertBlock(0x0, 0xc001f59560, 0x1)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:148 +0x2bd
github.com/tmccombs/hcl2json/convert.(*converter).convertBody(0x0, 0xc000611760)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:80 +0x9b
github.com/tmccombs/hcl2json/convert.ConvertFile(0x0, {0xc0})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:68 +0x6f
github.com/tmccombs/hcl2json/convert.File(0x0, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:37 +0x1d
github.com/tmccombs/hcl2json/convert.Bytes({0xc002465600, 0xc000a1c4e0, 0x2935860}, {0x0, 0x200}, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:27 +0x78
github.com/aquasecurity/fanal/config/parser/hcl2.Unmarshal({0xc002, 0xc0001ed0c0, 0xc0001ed0c0}, {0x22454a0, 0xc000159920})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220129174924-b9e05fcccc57/config/parser/hcl2/hcl2.go:11 +0x33
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.analyze({0x0}, {{0x0, 0x0}, {0xc007a2c600, 0x3c}, {0x2973140, 0xc007a362a0}, {0xc18c1d8, 0xc0001ed0c0}, {0x0}})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220129174924-b9e05fcccc57/analyzer/config/hcl/hcl.go:61 +0xdf
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.Analyze({0x3784ec8}, {0x2966618, 0xc00073f080}, {{0x0, 0x0}, {0xc007a2c600, 0x3c}, {0x2973140, 0xc007a362a0}, {0xc18c1d8, ...}, ...})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220129174924-b9e05fcccc57/analyzer/config/hcl/hcl.go:36 +0x6b
github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile.func1({0x2968410, 0x0}, {0x2968528, 0xc0001ed0c0})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220129174924-b9e05fcccc57/analyzer/analyzer.go:253 +0x275
created by github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220129174924-b9e05fcccc57/analyzer/analyzer.go:248 +0x3d2
```</div>
</details>

Getting exact same issue:

```
2022-02-28T17:22:37.054+0300	DEBUG	Missing image ID: sha256:339c839e3dd6dc0dda517a78c5bd392b39421a4a00e1282a972a88e8654c42b3
2022-02-28T17:22:37.054+0300	DEBUG	Missing diff ID: sha256:0a685305690c6b3534ffa345d962f137ec5e2dae50520f6c4f28ac18bf898459
2022-02-28T17:22:37.054+0300	DEBUG	Missing diff ID: sha256:ad081cb6688794ff475ecee8a9c97ba436b2dda38e500f71404eaac119fe7194
2022-02-28T17:23:07.449+0300	DEBUG	Analysis error: unable to decode TOML (go/pkg/mod/github.com/!burnt!sushi/toml@v0.3.1/_examples/hard.toml): toml: line 16 (last key "the.hard"): expected '.' or ']' to end table name, but got '#' instead
2022-02-28T17:23:07.449+0300	DEBUG	Analysis error: unable to decode TOML (go/pkg/mod/github.com/!burnt!sushi/toml@v0.3.1/_examples/invalid-apples.toml): toml: line 5 (last key "fruit"): Key 'fruit.type' has already been defined.
2022-02-28T17:23:07.478+0300	DEBUG	Parsing Java artifacts...	{"file": "go/pkg/mod/github.com/census-instrumentation/opencensus-proto@v0.2.1/gradle/wrapper/gradle-wrapper.jar"}
2022-02-28T17:23:07.579+0300	DEBUG	Analysis error: unable to decode JSON (go/src/.go/pkg/mod/github.com/go-playground/universal-translator@v0.17.0/testdata/bad-translation6.json): invalid character ']' after object key:value pair
2022-02-28T17:23:07.615+0300	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/array_comment_2.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/array_comment_2.hcl): convert hcl2 to bytes: parse config: [:4,5-6: Missing item separator; Expected a comma to mark the beginning of the next item.]
	* unable to parse HCL1 (go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/array_comment_2.hcl): At 4:5: error parsing list, expected comma or list end, got: STRING


2022-02-28T17:23:07.616+0300	DEBUG	Analysis error: unable to parse HCL (%!a(string=go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/git_crypt.hcl)): 2 errors occurred:
	* unable to parse HCL2 (go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/git_crypt.hcl): convert hcl2 to bytes: parse config: [:1,1-2: Invalid character; This character is not used within the language. :1,1-2: Argument or block definition required; An argument or block definition is required here.]
	* unable to parse HCL1 (go/src/.go/pkg/mod/github.com/hashicorp/hcl@v1.0.0/hcl/parser/test-fixtures/git_crypt.hcl): At 1:1: unexpected null character (0x00)

...
...
...
```

Eventually it panics:

```
panic: interface conversion: interface {} is convert.jsonObj, not []interface {}

goroutine 701 [running]:
github.com/tmccombs/hcl2json/convert.(*converter).convertBlock(0x0, 0xc0029090e0, 0x1)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:148 +0x2bd
github.com/tmccombs/hcl2json/convert.(*converter).convertBody(0x0, 0xc0016c1130)
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:80 +0x9b
github.com/tmccombs/hcl2json/convert.ConvertFile(0x0, {0x60})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:68 +0x6f
github.com/tmccombs/hcl2json/convert.File(0x0, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:37 +0x1d
github.com/tmccombs/hcl2json/convert.Bytes({0xc0006c4200, 0x0, 0x2d26fe0}, {0x0, 0x200}, {0x69})
	/home/runner/go/pkg/mod/github.com/tmccombs/hcl2json@v0.3.3/convert/convert.go:27 +0x78
github.com/aquasecurity/fanal/config/parser/hcl2.Unmarshal({0xc0006c4200, 0xc002bea1b0, 0xc002bea1b0}, {0x25ba820, 0xc002487e20})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220225095822-ef150f781751/config/parser/hcl2/hcl2.go:11 +0x33
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.analyze({0x14}, {{0x0, 0x0}, {0xc002324820, 0x4f}, {0x2d69448, 0xc000840620}, {0xa539058, 0xc002bea1b0}, {0x0}})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220225095822-ef150f781751/analyzer/config/hcl/hcl.go:61 +0xdf
github.com/aquasecurity/fanal/analyzer/config/hcl.ConfigAnalyzer.Analyze({0x616d6d7573222020}, {0x6f5222203a227972, 0x656720746547656c}, {{0x0, 0x0}, {0xc002324820, 0x4f}, {0x2d69448, 0xc000840620}, {0xa539058, ...}, ...})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220225095822-ef150f781751/analyzer/config/hcl/hcl.go:36 +0x6b
github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile.func1({0x2d5bf00, 0x0}, {0x2d5c018, 0xc002bea1b0})
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220225095822-ef150f781751/analyzer/analyzer.go:253 +0x275
created by github.com/aquasecurity/fanal/analyzer.AnalyzerGroup.AnalyzeFile
	/home/runner/go/pkg/mod/github.com/aquasecurity/fanal@v0.0.0-20220225095822-ef150f781751/analyzer/analyzer.go:248 +0x3d2
```


```
Version: 0.24.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-02-28 12:06:32.118785842 +0000 UTC
  NextUpdate: 2022-02-28 18:06:32.118785342 +0000 UTC
  DownloadedAt: 2022-02-28 14:12:31.549743 +0000 UTC
```

--- a/go.mod
+++ b/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/NYTimes/gziphandler v0.0.0-20170623195520-56545f4a5d46
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
-	github.com/aquasecurity/fanal v0.0.0-20220303080309-254063f95ea0
+	github.com/aquasecurity/fanal v0.0.0-20220303131110-677831c145ce
 	github.com/aquasecurity/go-dep-parser v0.0.0-20220302151315-ff6d77c26988
 	github.com/aquasecurity/go-gem-version v0.0.0-20201115065557-8eed6fe000ce
 	github.com/aquasecurity/go-npm-version v0.0.0-20201110091526-0b796d180798
@@ -73,8 +73,8 @@ require (
 	github.com/agext/levenshtein v1.2.3 // indirect
 	github.com/apparentlymart/go-cidr v1.1.0 // indirect
 	github.com/apparentlymart/go-textseg/v13 v13.0.0 // indirect
-	github.com/aquasecurity/defsec v0.12.1 // indirect
-	github.com/aquasecurity/tfsec v1.4.1 // indirect
+	github.com/aquasecurity/defsec v0.14.0 // indirect
+	github.com/aquasecurity/tfsec v1.5.0 // indirect
 	github.com/aws/aws-sdk-go v1.43.8 // indirect
 	github.com/bgentry/go-netrc v0.0.0-20140422174119-9fd32a8b3d3d // indirect
 	github.com/bmatcuk/doublestar v1.3.4 // indirect
@@ -152,7 +152,7 @@ require (
 	github.com/sergi/go-diff v1.1.0 // indirect
 	github.com/sirupsen/logrus v1.8.1 // indirect
 	github.com/stretchr/objx v0.3.0 // indirect
-	github.com/tmccombs/hcl2json v0.3.3 // indirect
+	github.com/tmccombs/hcl2json v0.3.4 // indirect
 	github.com/ulikunitz/xz v0.5.8 // indirect
 	github.com/vbatts/tar-split v0.11.2 // indirect
 	github.com/xanzy/ssh-agent v0.3.0 // indirect

